py/desispec/fluxcalibration.py

```diff
@@ -850,23 +850,32 @@ def compute_flux_calibration(frame, input_model_wave,input_model_flux,input_mode
         convolved_model_flux[fiber]=stdstars.R[fiber].dot(model_flux[fiber])
 
     # iterative fitting and clipping to get precise mean spectrum
-    current_ivar=stdstars.ivar.copy()
-
+    current_ivar=stdstars.ivar*(stdstars.mask==0)
+    
     #- Start with a first pass median rejection
     calib = (convolved_model_flux!=0)*(stdstars.flux/(convolved_model_flux + (convolved_model_flux==0)))
     median_calib = np.median(calib, axis=0)
-    chi2 = stdstars.ivar * (stdstars.flux - convolved_model_flux*median_calib)**2
-    bad=(chi2>nsig_clipping**2)
-    current_ivar[bad] = 0
 
+    # First fit of smooth correction per fiber, and 10% model error to variance,  and perform first outlier rejection
     smooth_fiber_correction=np.ones((stdstars.flux.shape))
-    chi2=np.zeros((stdstars.flux.shape))
-
-    # chi2 = sum w ( data_flux - R*(calib*model_flux))**2
-    # chi2 = sum (sqrtw*data_flux -diag(sqrtw)*R*diag(model_flux)*calib)
+    chi2=np.zeros((stdstars.flux.shape))  
 
+    for fiber in range(nstds) :
+        M = median_calib*stdstars.R[fiber].dot(model_flux[fiber])
+        
+        try:
+            pol=np.poly1d(np.polyfit(dwave,stdstars.flux[fiber]/(M+(M==0)),deg=deg,w=current_ivar[fiber]*M**2))
+        except:
```

Comment:
What's the specific exception that you might encounter here?   perhaps np.linalg.LinAlgError?  Please catch that and only that so that any other unexpected exceptions still crash.